src/pybind/mgr/cephadm/tests/test_scheduling.py

```diff
@@ -1697,3 +1703,42 @@ def test_drain_from_explict_placement(service_type, placement, hosts, maintenanc
     ).place()
     assert sorted([h.hostname for h in to_add]) in expected_add
     assert sorted([h.name() for h in to_remove]) in expected_remove
+
+
+class RegexHostPatternTest(NamedTuple):
+    service_type: str
+    placement: PlacementSpec
+    hosts: List[str]
+    expected_add: List[List[str]]
+
+
+@pytest.mark.parametrize("service_type,placement,hosts,expected_add",
```

Comment:
I kinda think this next block would look a lot nicer if it were 'black'-formatted, but it's fine I suppose. :-)